tests/fv/kdd_node_ipam_test.go

```diff
@@ -0,0 +1,348 @@
+			handleA := "handleA"
+			attrs := map[string]string{"node": nodeA, "pod": "pod-a", "namespace": "default"}
+			err = calicoClient.IPAM().AssignIP(context.Background(), ipam.AssignIPArgs{
+				IP: net.MustParseIP("192.168.0.1"), HandleID: &handleA, Attrs: attrs, Hostname: nodeA,
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			// Allocate a pod IP address and thus a block and affinity to NodeB.
+			handleB := "handleB"
+			attrs = map[string]string{"node": nodeB, "pod": "pod-b", "namespace": "default"}
+			err = calicoClient.IPAM().AssignIP(context.Background(), ipam.AssignIPArgs{
+				IP: net.MustParseIP("192.168.0.65"), HandleID: &handleB, Attrs: attrs, Hostname: nodeB,
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			// Allocate a pod IP address and thus a block and affinity to NodeC.
+			handleC := "handleC"
+			attrs = map[string]string{"node": nodeC, "pod": "pod-c", "namespace": "default"}
+			err = calicoClient.IPAM().AssignIP(context.Background(), ipam.AssignIPArgs{
+				IP: net.MustParseIP("192.168.0.129"), HandleID: &handleC, Attrs: attrs, Hostname: nodeC,
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			// Release the affinity for the block, creating the desired state - an IP address in a non-affine block.
+			err = calicoClient.IPAM().ReleaseHostAffinities(context.Background(), nodeC, false)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Also allocate an IP address on NodeC within NodeB's block, to simulate a "borrowed" address.
+			handleC2 := "handleC2"
+			attrs = map[string]string{"node": nodeC, "pod": "pod-c2", "namespace": "default"}
+			err = calicoClient.IPAM().AssignIP(context.Background(), ipam.AssignIPArgs{
+				IP: net.MustParseIP("192.168.0.66"), HandleID: &handleC2, Attrs: attrs, Hostname: nodeC,
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			// Expect the correct blocks to exist as a result of the IPAM allocations above.
+			blocks, err := bc.List(context.Background(), model.BlockListOptions{}, "")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(blocks.KVPairs)).To(Equal(3))
+			affs, err := bc.List(context.Background(), model.BlockAffinityListOptions{Host: nodeA}, "")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(affs.KVPairs)).To(Equal(1))
+			affs, err = bc.List(context.Background(), model.BlockAffinityListOptions{Host: nodeB}, "")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(affs.KVPairs)).To(Equal(1))
+			affs, err = bc.List(context.Background(), model.BlockAffinityListOptions{Host: nodeC}, "")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(affs.KVPairs)).To(Equal(0))
+
+			// Deleting NodeB should clean up the allocations associated with the node, as well as the
+			// affinity, but should leave the block intact since there are still allocations from another
+			// node.
+			err = k8sClient.CoreV1().Nodes().Delete(nodeB, nil)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(func() error {
+				if err := assertIPsWithHandle(calicoClient.IPAM(), handleB, 0); err != nil {
```

Comment:
Can you include positive checks to make sure handleA and handleC still have their IPs at this stage?  I.e. the cleanup didn't have any collateral damage?